src/normalizer/nonspacing_mark.rs

```diff
@@ -0,0 +1,162 @@
+use std::borrow::Cow;
+use std::collections::HashSet;
+
+use once_cell::sync::Lazy;
+
+use super::{Normalizer, NormalizerOption};
+use crate::detection::{Language, Script};
+use crate::Token;
+
+static HASHSET: Lazy<HashSet<u32>> = Lazy::new(|| {
+    let bytes = include_bytes!("../../dictionaries/bin/nonspacing_mark/marks.bin");
+
+    HashSet::from_iter(
+        bytes.chunks_exact(4).map(|chunk| u32::from_ne_bytes(chunk.try_into().unwrap())),
+    )
+});
+
+/// A global [`Normalizer`] removing nonspacing marks.
+///
+/// This normalizer uses built-in `HashSet` internally to check over the marks set
+pub struct NonspacingMarkNormalizer;
+
+impl Normalizer for NonspacingMarkNormalizer {
+    fn normalize<'o>(
+        &self,
+        mut token: Token<'o>,
+        options: NormalizerOption,
+    ) -> Box<dyn Iterator<Item = Token<'o>> + 'o> {
+        if token.lemma().chars().any(is_nonspacing_mark) {
+            let mut lemma = String::new();
+            let mut char_map = if options.create_char_map { Some(Vec::new()) } else { None };
+
+            for c in token.lemma().chars() {
+                if is_nonspacing_mark(c) {
+                    char_map.as_mut().map(|char_map| char_map.push((c.len_utf8() as u8, 0)));
+                } else {
+                    char_map
+                        .as_mut()
+                        .map(|char_map| char_map.push((c.len_utf8() as u8, c.len_utf8() as u8)));
+
+                    lemma.push(c);
+                }
+            }
+
+            token.lemma = Cow::Owned(lemma);
+            token.char_map = char_map;
+        }
+
+        Box::new(Some(token).into_iter())
+    }
+
+    fn should_normalize(&self, _script: Script, _language: Option<Language>) -> bool {
+        true
+    }
```

Comment:
To avoid potential performance drawbacks, we could activate this Normalizer only for a subset of Languages, then we extend this list depending on future feedback. 🤔 
```suggestion
    fn should_normalize(&self, script: Script, _language: Option<Language>) -> bool {
        matches!(script, Script::Hebrew | Script::Thai | Script::Arabic)
    }
```